app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/pickfiles/components/EmailAddressTextField.kt

```diff
@@ -152,14 +159,26 @@ private class EmailAddressTextFieldState(
         }
     }
 
+    fun getFirstText(clipboardManager: ClipboardManager): String? {
+        val countItemInClipboard = clipboardManager.primaryClip?.itemCount ?: return null
+        val description = clipboardManager.primaryClipDescription ?: return null
+
+        (0 until countItemInClipboard).forEach { index ->
+            if (description.getMimeType(index).startsWith("text/")) {
+                val text = clipboardManager.primaryClip?.getItemAt(index)?.text?.toString()
+                if (text?.isNotBlank() == true) return text
+            }
+        }
+        return null
+    }
```

Comment:
You should avoid duplicating logic like this and instead find a way to share the logic across all places that need it